webserver/static/scripts/datasets/editor.js

```diff
@@ -460,39 +462,56 @@ var RecordingAddForm = React.createClass({
     handleSubmit: function (event) {
         event.preventDefault();
         var mbid = this.refs.mbid.value.trim();
+        mbid = RECORDING_MBID_RE.exec(mbid)[2];
         if (!mbid) {
             return;
         }
+        mbid = mbid.toLowerCase();
         this.props.onRecordingSubmit(mbid);
         this.refs.mbid.value = '';
     },
     handleChange: function () {
-        var mbid = this.refs.mbid.value;
-        var isValidUUID = /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i.test(mbid);
-        var isNotDuplicate = this.props.recordings.indexOf(mbid) == -1;
+        var mbidField = this.refs.mbid.value;
+        var isValidUUID = RECORDING_MBID_RE.test(mbidField);
+        var isNotDuplicate = false;
```

Comment:
`isNotDuplicate` seems like a weird bool variable to read, maybe replacing it with `isDuplicate` would be a good idea.

Would also make reading this condition in the code easier:

`else if (this.state.length > 0 && !this.state.notDuplicate) {` 

vs

`else if (this.state.length > 0 && this.state.isDuplicate) {`

Reply:
the biggest reason is because it was like that to begin with :) - it made sense when the only condition it was being used in was `validInput: isValidUUID && isNotDuplicate`, but you're right. Changing it to isDuplicate should make it easier to reason about in all of the places it's used

Reply:
I fixed it (had the time) :P